<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigWidget</class>
 <widget class="QWidget" name="ConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>685</width>
    <height>627</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Build Directory:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="KUrlRequester" name="buildDir">
       <property name="toolTip">
        <string>Select the directory where the tools are being executed in.
If this is the same as the source directory you can leave it empty.</string>
       </property>
       <property name="mode">
        <set>KFile::Directory|KFile::ExistingOnly|KFile::LocalOnly</set>
       </property>
       <property name="clickMessage">
        <string>The directory where the tools are executed</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Build Tools</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Action:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KComboBox" name="buildAction"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Enable:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="enableAction">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="execLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Executable:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="KUrlRequester" name="actionExecutable">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>This is the executable that should be executed when selecting the action from the Project menu</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="argLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Arguments:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="actionArguments">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The arguments that the tool needs to execute properly.</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="envLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Environment:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="KDevelop::EnvironmentSelectionWidget" name="actionEnvironment">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Choose a pre-configured environment for your tool</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Path-specific language parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="9" column="0">
       <widget class="KTabWidget" name="languageParameters">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>8</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="includePathsPage">
         <attribute name="title">
          <string>Includes/Imports</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="KUrlRequester" name="includePathRequester">
              <property name="clickMessage">
               <string>A new include path to use for parsing files under given directory</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="KPushButton" name="addIncludePath">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Create a new include path entry from this URL.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="KPushButton" name="saveIncludePath">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Replace selected include path entry with this URL.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Replace</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QListView" name="includePaths">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>8</verstretch>
             </sizepolicy>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="definesPage">
         <attribute name="title">
          <string>Defines</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTableView" name="defines">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>8</verstretch>
             </sizepolicy>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="8" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="KComboBox" name="projectPaths">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pick a sub-project path to configure language parameters for.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KPushButton" name="removePath">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Delete current sub-project path entry.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="KUrlRequester" name="pathRequester">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Modify/enter a URL to use it in the list below.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Note:&lt;/span&gt; the URL must belong to the project.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KPushButton" name="addPath">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Create a new sub-project path entry from this URL.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KPushButton" name="savePath">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Replace selected sub-project path entry with this URL.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Replace</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KPushButton</class>
   <extends>QPushButton</extends>
   <header>kpushbutton.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KTabWidget</class>
   <extends>QTabWidget</extends>
   <header>ktabwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentSelectionWidget</class>
   <extends>QComboBox</extends>
   <header>util/environmentselectionwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionExecutable</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionArguments</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionEnvironment</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>execLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>argLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>35</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>envLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>34</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
